source/common/http/http1/codec_impl.h

```diff
@@ -311,6 +318,7 @@ class ConnectionImpl : public virtual Connection,
   bool dispatching_ : 1;
   bool dispatching_slice_already_drained_ : 1;
   const bool no_chunked_encoding_header_for_304_ : 1;
+  StreamInfo::BytesMeter bytes_meter_before_stream_;
```

Comment:
This is almost what I'm suggesting. I think you can make this even simpler by:
1) Make this a shared_ptr, and allocate it on construction.
2) When you have a stream, just move the allocated meter to the stream, taking the accounting with it.
3) Allocate a new connection level one the next time you need it.
4) Repeat.

Reply:
It is difficult to move the bytes meter to the stream here, we only have const reference of bytes_meter_ member inside the ConnectionImpl

Reply:
Sorry I think I'm not being clear.

1) make_shared at the connection
2) literally move the shared object to the stream
3) Repeat. 

You don't need any reference object at the connection level. This will make all of the accounting much simpler.

/wait